girder/api/v1/folder.py

```diff
@@ -179,22 +179,38 @@ def updateFolder(self, folder, params):
     @loadmodel(model='folder', level=AccessType.ADMIN)
     def updateFolderAccess(self, folder, params):
         self.requireParams('access', params)
+        user = self.getCurrentUser()
 
         public = self.boolParam('public', params, default=False)
-        self.model('folder').setPublic(folder, public)
+        recurse = self.boolParam('recurse', params, default=False)
+        progress = self.boolParam('progress', params, default=False) and recurse
 
         try:
             access = json.loads(params['access'])
-            return self.model('folder').setAccessList(
-                folder, access, save=True)
         except ValueError:
             raise RestException('The access parameter must be JSON.')
+
+        with ProgressContext(progress, user=user, title='Updating permissions',
+                             message='Calculating progress...') as ctx:
+            if progress:
+                ctx.update(total=self.model('folder').subtreeCount(
+                    folder, includeItems=False, user=user,
+                    level=AccessType.ADMIN))
+            return self.model('folder').setAccessList(
+                folder, access, save=True, recurse=recurse, user=user,
+                progress=ctx, setPublic=public)
+
     updateFolderAccess.description = (
         Description('Update the access control list for a folder.')
         .param('id', 'The ID of the folder.', paramType='path')
         .param('access', 'The JSON-encoded access control list.')
         .param('public', "Whether the folder should be publicly visible.",
                dataType='boolean')
+        .param('recurse', 'Whether the policies should be applied to all '
+               'subfolders under this folder as well (default=False).',
+               dataType='boolean')
+        .param('progress', 'If recurse is set to True, this controls whether '
```

Comment:
also can you add the default values here to progress and public?

2 additional notes that don't require any action.

1) This method is quite similar to the same one in the collection API.  They *could* be joined up, but that would take more work, and prevent us from having separate behavior in the future for the two separate resources.

2) Wouldn't it be cool to be able to set the default value for a param in the description **xor** the code, and have the one read from the other?